pkg/math/oracle/median.go

```
@@ -33,279 +41,166 @@ func NewMedianAggregator(logger *zap.Logger, cfg mmtypes.MarketMap) (*MedianAggr
 		}
 
-		aggregatedMedians[ticker] = unscaledPrice
+		// Take the median of the converted prices. This takes the average of the middle two
+		// prices if the number of prices is even.
+		price := median.CalculateMedian(convertedPrices)
+		updatedPrices[target] = price
+		m.logger.Info(
+			"calculated median price",
+			zap.String("ticker", ticker),
+			zap.String("price", price.String()),
+			zap.Any("converted_prices", convertedPrices),
+		)
+
 	}
 
-	return aggregatedMedians
+	// Update the aggregated data. These prices are going to be used as the index prices the
+	// next time we calculate prices.
+	m.PriceAggregator.SetAggregatedData(updatedPrices)
+	m.logger.Info("calculated median prices for price feeds", zap.Int("num_prices", len(updatedPrices)))
 }
 
-// CalculateConvertedPrices calculates the converted prices for each ticker using the
-// provided median prices and the conversion markets.
-//
-// For example, if the oracle receives a price for BTC/USDT and USDT/USD, it can use the conversion
-// market to convert the BTC/USDT price to BTC/USD. In this case, the medians map would contain
-// the median prices for BTC/USDT and USDT/USD, and the conversions would contain a sorted list of
-// operations to convert the price of BTC/USDT to BTC/USD i.e. BTC/USDT * USDT/USD = BTC/USD.
+// CalculateConvertedPrices calculates the converted prices for a given set of paths and target ticker.
+// The prices utilized are the prices most recently seen by the providers. Each price is within a
+// MaxPriceAge window so is safe to use.
 func (m *MedianAggregator) CalculateConvertedPrices(
-	ticker mmtypes.Ticker,
+	target mmtypes.Ticker,
 	paths mmtypes.Paths,
-	medians types.TickerPrices,
 ) []*big.Int {
-	convertedPrices := make([]*big.Int, 0)
+	m.logger.Debug("calculating converted prices", zap.String("ticker", target.String()))
+	if len(paths.Paths) == 0 {
+		m.logger.Error(
+			"no conversion paths",
+			zap.String("ticker", target.String()),
+		)
+
+		return nil
```

Is this an error or a valid scenario?

We are logging an error but returning `nil`.  If we want to handle this error gracefully, we should return a typed error and then have the calling function handle that.